dotnet/src/Microsoft.Agents.AI.Workflows/WorkflowThread.cs

```diff
@@ -153,13 +153,27 @@ IAsyncEnumerable<AgentRunResponseUpdate> InvokeStageAsync(
                     case AgentRunUpdateEvent agentUpdate:
                         yield return agentUpdate.Update;
                         break;
+
                     case RequestInfoEvent requestInfo:
                         FunctionCallContent fcContent = requestInfo.Request.ToFunctionCall();
                         AgentRunResponseUpdate update = this.CreateUpdate(this.LastResponseId, fcContent);
                         yield return update;
                         break;
+
                     case SuperStepCompletedEvent stepCompleted:
                         this.LastCheckpoint = stepCompleted.CompletionInfo?.Checkpoint;
+                        goto default;
+
+                    default:
+                        // Emit all other workflow events for observability (DevUI, logging, etc.)
+                        yield return new AgentRunResponseUpdate(ChatRole.Assistant, [])
+                        {
+                            CreatedAt = DateTimeOffset.UtcNow,
+                            MessageId = Guid.NewGuid().ToString("N"),
+                            Role = ChatRole.Assistant,
+                            ResponseId = this.LastResponseId,
+                            RawRepresentation = evt
+                        };
```

Comment:
cc @lokitoth Does this sound like a reasonable thing to do, i.e, emitting workflow events in this way for observability.

Reply:
Yes, that seems reasonable - and in line with what we discussed about emitting WorkflowEvents through the Agent APIs earlier; I wonder if we can make some automated way of converting appropriate events to AIContent.